main.go

```diff
@@ -174,7 +175,14 @@ func generateEngine() *gin.Engine {
 		panic(err)
 	}
 
-	r := gin.Default()
+	// Initalize Gin
+	gin.SetMode(gin.ReleaseMode)
+	// Instantiate a new engine
+	r := gin.New()
+	// Use our custom logger
+	r.Use(middleware.DefaultStructuredLogger())
```

Comment:
this naming "default" seems weird - as it's our own custom middleware and not the default

the middle layer makes sense as a shim to inject the logger object - that's all good, just naming

Reply:
maybe we name this function "StructuredLogger" and leave the underlying function private? e.g. in python i would name the lower level one "_StructuredLogger"

Reply:
maybe there is a recommended pattern for this in gin